packages/story-editor/src/app/story/useStoryReducer/reducers/duplicateElementsById.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { duplicateElement } from '../../../../elements';
+
+/**
+ * Duplicate all elements specified by `elementIds` on the current page.
+ * Set selected elements to be the newly created elements.
+ *
+ * If given `elementIds` are not a list, do nothing.
+ *
+ * @param {Object} state Current state
+ * @param {Object} payload Action payload
+ * @param {Array.<string>} payload.elementIds Array of ids of elements to duplicate.
+ * @return {Object} New state
+ */
+function duplicateElementsById(state, { elementIds }) {
+  if (!Array.isArray(elementIds)) {
+    return state;
+  }
+
+  const pageIndex = state.pages.findIndex(({ id }) => id === state.current);
+  const oldPage = state.pages[pageIndex];
+
+  const newPage = { ...oldPage };
+  const newSelection = [];
+
+  elementIds.forEach((elementId) => {
+    const elementIndex = newPage.elements.findIndex(
+      ({ id }) => id === elementId
+    );
+
+    if (elementIndex < 0) {
+      return;
+    }
+
+    const elementToDuplicate = newPage.elements[elementIndex];
+
+    if (elementToDuplicate.isBackground) {
+      return;
+    }
+
+    const { element, elementAnimations } = duplicateElement({
+      element: elementToDuplicate,
+      animations: oldPage.animations,
+      existingElements: oldPage.elements,
+    });
+
+    newSelection.push(element.id);
+
+    newPage.animations = [...(newPage.animations || []), ...elementAnimations];
+    newPage.elements = [
```

Comment:
One thing that happens like this is that the duplicated layer is added right next to the origin element, e.g. if I duplicate 3 elements:
- IMAGE 1
- IMAGE 2
- IMAGE 3

it becomes
- IMAGE 1
- IMAGE 1 dupe
- IMAGE 2
- IMAGE 2 dupe
- IMAGE 3
- IMAGE 3 dupe

instead of 
- IMAGE 1
- IMAGE 2
- IMAGE 3
- IMAGE 1 dupe
- IMAGE 2 dupe
- IMAGE 3 dupe

Is this intended? I was thinking that all the dupes would go to the end of the list of the layers as new elements, however, not really sure what's the standard behaviour in this case for layers. Thoughts?

Reply:
Good question. I don't have context as to why that happens, but this is what existed before this ticket:

https://github.com/GoogleForCreators/web-stories-wp/blob/main/packages/story-editor/src/app/story/useStoryReducer/reducers/duplicateElementById.js#L47-L52

@littlemilkstudio do you have more context here?

Reply:
This should be a question for UX not eng - @agingoldseco  do you have a quick second to weigh in on this?

Reply:
@agingoldseco this pr fixes that! More wondering about placement of the newly created elements.

Seems like in figma that they're all added to the end/beginning of the layers together rather than splitting them up and placing them directly before their original element.

Reply:
@agingoldseco this pr leaves the newly created elements selected 👍 

Seems like figma also places the newly created elements right next to their parent elements. For now I'll leave the behavior as is, but it will be an easy change if in the future we'd like that to act the same as copy/paste.